### PR TITLE
HTTPClient refactor

### PR DESCRIPTION
**Pull request by kingoflolz**
- remove threadpool (make everything async)
- Add exit requests to it
- Use Actix client everywhere

### CODE REVIEW

File: rita/Cargo.toml [hunk 1]
```diff
@@ -25,8 +25,8 @@ exit_db = { path = "../exit_db" }
 num256 = { path = "../num256" }
 settings = { path = "../settings" }
 
-actix = "0.5.7"
-actix-web = { version = "0.6.7", default-features = false}
+actix = "0.5.8"
+actix-web = { version = "0.6.12", default-features = false}
```
Comment (jkilpatr) [inline rita/Cargo.toml]: out of curiosity what did you upgrade for?

File: rita/src/rita_client/exit_manager/mod.rs [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
+use althea_types::{ExitClientDetails, ExitClientIdentity, ExitState};
 
 use settings::{RitaClientSettings, RitaCommonSettings};
 use SETTING;
 
 use rita_client::rita_loop::Tick;
 use rita_client::traffic_watcher::{TrafficWatcher, Watch};
+use rita_common::http_client::{ExitSetupRequest, GetExitInfo, HTTPClient};
+
+use futures::Future;
```
Comment (jkilpatr) [inline rita/src/rita_client/exit_manager/mod.rs]: so these are native futures as opposed to actix futures now?

File: scripts/bridge.sh [hunk 3]
```diff
@@ -1,8 +1,8 @@
 #!/usr/bin/env bash
 # this bridges a router into the integration test mesh
 
-ip link set enx000ec6a0b495 netns netlab-1
-ip netns exec netlab-1 ifconfig enx000ec6a0b495 up
+# ip link set enx000ec6a0b495 netns netlab-1
```
Comment (jkilpatr) [inline scripts/bridge.sh]: what's going on here?

File: rita/Cargo.toml (hunk 1 above)
Comment (kingoflolz) [reply rita/Cargo.toml]: It had some new features for the http client which made this possible

File: rita/src/rita_client/exit_manager/mod.rs (hunk 2 above)
Comment (kingoflolz) [reply rita/src/rita_client/exit_manager/mod.rs]: they are native futures as opposed to doing blocking operations within actors

**Pull request merged**